sdk/messaging/azeventhubs/internal/links.go

```diff
@@ -264,7 +325,8 @@ func (l *Links[LinkT]) newLinkState(ctx context.Context, partitionID string) (*l
 	session, connID, err := l.ns.NewAMQPSession(ctx)
 
 	if err != nil {
-		_ = ls.Close(ctx)
+		azlog.Writef(exported.EventConn, "(%s): Failed to create AMQP session for partition ID '%s': %s", ls.String(), partitionID, err)
+		err = ls.Close(ctx)
```

Comment:
This will overwrite the last error.  I'm guessing this is intentional but wanted to confirm.

Reply:
I believe it's unintentional - I think we want the original error, not the Close() error. Will switch.